### ISSUE TITLE
`set_array` into array does and into object does not work

### ISSUE THREAD

**Issue by TobiasNx** [metafacture/metafacture-fix#130]
Found by: @acka47 

```
set_array("animals[].*.test[]")

set_array("animals[].$last.TEST[]")

set_array("animals[].$append.T__ST[]")
```

Results in:
```
{
  "animals" : [ {
    "name" : "Jake",
    "animalGroup" : "dog"
  }, {
    "name" : "Blacky",
    "animalGroup" : "parrot"
  } ],
  "animals[].*.test" : [ ],
  "animals[].$last.TEST" : [ ],
  "animals[].$append.T__ST" : [ ]
}
```

Output expected:
```
{
	"animals": [
		{
			"name": "Jake",
			"animalGroup": "dog",
			"test":[]
		},
		{
			"name": "Blacky",
			"animalGroup": "parrot",
			"test":[],
			"TEST":[]
		},
		{
			"T__ST":[]
		}
	]
}
```

See test scenario here: https://github.com/TobiasNx/fix-FunctionalReview-Testing/tree/master/data/testing/setArrayIntoArray

**Comment by fsteeg** [comment]
Consider in / add as integration test / revisit after #102.

**Comment by TobiasNx** [comment]
has integration tests: https://github.com/metafacture/metafacture-fix/commit/b1a7336aea600ccea7a2e9d8e9eccaaa55c8c1eb
